test/controllers/auctions_controller_test.rb

```diff
@@ -0,0 +1,76 @@
+require 'test_helper'
+require 'date'
+
+class AuctionsControllerTest < ActionDispatch::IntegrationTest
+  test 'should get /auctions endpoint' do
+    get '/auctions'
+    assert_response :success
+  end
+
+  test '/auctions should retrieve all auctions as json' do
+    auctions_to_create = [
+      {
+        title: 'Mustang',
+        description: '1968 Ford Mustang',
+        start_bid: 12000.00,
+        current_bid: 12000.00,
+        end_time: 10.days.from_now,
+      },
+      {
+        title: 'Fallen Empires Booster',
+        description: '3x MTG Fallen Empires booster packs',
+        start_bid: 10.00,
+        current_bid: 14.50,
+        end_time: 10.days.from_now,
+      }
+    ]
+
+    auctions_to_create.each do |auction| Auction.create(
+        title: auction[:title],
+        description: auction[:description],
+        start_bid: auction[:start_bid],
+        current_bid: auction[:current_bid],
+        end_time: auction[:end_time]
+      ) end
+
+    get '/auctions'
+    auction_response = JSON.parse(@response.body, symbolize_names: true)
+    response_auction_titles = auction_response.map {|auction| auction[:title]}
+    assert_equal 2, auction_response.length
+    assert_equal auctions_to_create.map {|auction| auction[:title]}, response_auction_titles
+  end
+
+  test 'should get /auctions/1 endpoint' do
+    created_auction = Auction.create(
+      title: 'Mustang',
+      description: '1968 Ford Mustang',
+      start_bid: 12000.00,
+      current_bid: 12000.00,
+      end_time: DateTime.now + 10
+    )
+    get "/auctions/#{created_auction.id}"
+    assert_response :success
+  end
+
+  test 'should return the correct auction given an id' do
+    auction_to_create = {
+      title: 'Mustang',
+      description: '1968 Ford Mustang',
+      start_bid: 12000.00,
+      current_bid: 12000.00,
+      end_time: 10.days.from_now.strftime('%FT%TZ')
+    }
+    created_auction = Auction.create(auction_to_create)
+
+    get "/auctions/#{created_auction[:id]}"
```

Comment:
Can we use dot syntax here?

Reply:
We can, but I'm honestly not certain why you'd chose dot syntax over grabbing the symbol. The whole 'dot syntax vs symbol' is still confusing to me, and it's hard for me to anticipate which will work in which instance.

Reply:
1 -- consistency, because that's how you do it above.
2 -- (and we can totally go over this more) re dot vs symbol: `auction.id` means you are accessing the `id` method on the Auction object, which returns the value of the `id` field in the DB. `auction[:id]` works *only* because Rails has added things that let you use brackets and the symbol name of the field to get its value.

If you had:

```ruby
class Dog
  def initialize(name, age)
    @age = age
    @name = name
  end

  def age
    @age
  end

  def name
    @name
  end
end
```

Then if you had `simon = Dog.new('Simon', 13.5)`, doing `simon.name` would return `'Simon'` but `simon[:name]` would blow up. The only reason you can do both in the tests and app is because Auction etc is an ActiveRecord model.